depthai_ros_driver/src/ipc_test.cpp

```diff
@@ -0,0 +1,130 @@
+#include <unistd.h>
+
+#include <chrono>
+#include <cinttypes>
+#include <cstdio>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <utility>
+
+#include "depthai_ros_driver/camera.hpp"
+#include "opencv2/highgui/highgui.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/image.hpp"
+#define GETPID getpid
+
+#include "builtin_interfaces/msg/time.hpp"
+#include "opencv2/opencv.hpp"
+
+int encoding2mat_type(const std::string& encoding) {
+    if(encoding == "mono8") {
+        return CV_8UC1;
+    } else if(encoding == "bgr8") {
+        return CV_8UC3;
+    } else if(encoding == "16UC1") {
+        return CV_16SC1;
+    } else if(encoding == "rgba8") {
+        return CV_8UC4;
+    }
+    throw std::runtime_error("Unsupported mat type");
+}
+
+std::string mat_type2encoding(int mat_type) {
+    switch(mat_type) {
+        case CV_8UC1:
+            return "mono8";
+        case CV_8UC3:
+            return "bgr8";
+        case CV_16SC1:
+            return "mono16";
+        case CV_8UC4:
+            return "rgba8";
+        default:
+            throw std::runtime_error("Unsupported encoding type");
+    }
+}
+
+void set_now(builtin_interfaces::msg::Time& time) {
+    std::chrono::nanoseconds now = std::chrono::high_resolution_clock::now().time_since_epoch();
+    if(now <= std::chrono::nanoseconds(0)) {
+        time.sec = time.nanosec = 0;
+    } else {
+        time.sec = static_cast<builtin_interfaces::msg::Time::_sec_type>(now.count() / 1000000000);
+        time.nanosec = now.count() % 1000000000;
+    }
+}
+
+void draw_on_image(cv::Mat& image, const std::string& text, int height) {
+    cv::Mat c_mat = image;
+    cv::putText(c_mat, text.c_str(), cv::Point(10, height), cv::FONT_HERSHEY_SIMPLEX, 0.3, cv::Scalar(0, 255, 0));
+}
+
+/// Node which receives sensor_msgs/Image messages and renders them using OpenCV.
+class ImageViewNode : public rclcpp::Node {
```

Comment:
Why not rqt_image_view ?

Reply:
This is specific for intra-process demo as it allows displaying memory value for given image data